spec/requests/sidebar_sections_controller_spec.rb

```diff
@@ -0,0 +1,163 @@
+# frozen_string_literal: true
+
+RSpec.describe SidebarSectionsController do
+  let(:admin) { Fabricate(:admin) }
+  fab!(:user) { Fabricate(:user) }
+
+  describe "#create" do
+    it "is not available for anonymous" do
+      post "/sidebar_sections.json",
+           params: {
+             title: "custom section",
+             links: {
+               "0" => {
```

Comment:
Will you be able to clarify what "0" means here?

Reply:
this is how params are represtented after this ajax call

```
    return ajax(`/sidebar_sections`, {
      type: "POST",
      data: {
        title: this.model.title,
        links: this.model.links.map((link) => {
          return {
            name: link.name,
            value: link.path,
          };
        }),
      },
    })
```

<img width="351" alt="Screenshot 2023-02-02 at 12 42 30 pm" src="https://user-images.githubusercontent.com/72780/216210575-88bc86bb-261b-4651-9505-88310291da7e.png">

Reply:
I believe you should be able to pass an array for `links`. Having to include the 'index" of the item in the params seems odd.

Reply:
Cannot find a way to make it nicely. jQuery ajax method is using $.param to serialize params

<img width="570" alt="Screenshot 2023-02-02 at 3 32 03 pm" src="https://user-images.githubusercontent.com/72780/216232204-1d576f4e-3960-4f11-b669-c3d19477af34.png">

It would not include index if I have an array of strings and not objects:
<img width="434" alt="Screenshot 2023-02-02 at 3 33 53 pm" src="https://user-images.githubusercontent.com/72780/216232414-65d69976-f24a-42b0-98b9-2e7869477d88.png">